### ISSUE TITLE
dkms support

### ISSUE THREAD

**Issue by Digitalone1** [morrownr/8821cu#19]
Hello, can you please provide instructions for dkms or make a specific dkms package on AUR? 

Thanks.

**Comment by morrownr** [comment]
What is the problem you are trying to solve?

Did you read the installation instructions? The instructions guide you through a dkms installation.

**Comment by Digitalone1** [comment]
No, sorry, I only read the package name on AUR without dkms word. My mistake. It's okay.

I'll try to install it. I have the 5.4 version. Do you recommend this 5.8? Is it better than 5.4?

**Comment by morrownr** [comment]
This driver works fine with Linux kernel 5.4 or 5.8. It works with all kernels up through 5.12.

It sounds like you are using Arch or Manjaro. I am not a regular user of either but I do have a test partition with Manjaro on it. Take care to install the correct headers. That is a source of problems with Manjaro.

Good luck.

**Comment by Digitalone1** [comment]
So, there's no real differences compared to [this](https://aur.archlinux.org/packages/rtl8821cu-dkms-git/) ([git](https://github.com/brektrou/rtl8821CU))?

Anyway, the [package](https://aur.archlinux.org/packages/rtl8821cu-git/) for your version on AUR it's not made for dkms installation.

Myabe I'll try to install it with a custom PKGBUILD, but a little suggestion: it's better `/kernel/drivers/net/wireless/realtek/rtl8821cu` as "dest module location" like is set [here](https://github.com/brektrou/rtl8821CU/blob/master/dkms.conf).

**Comment by morrownr** [comment]
Quote: "So, there's no real differences compared to this (git)?"

Reply: It looks like there is a big difference in that the git you are pointing to is based on a much older version of the driver.

Quote: "Anyway, the package for your version on AUR it's not made for dkms installation."

Reply: I see. I did not post that on AUR. I support the driver as it is right here in this repo. What others do with the source is up to them.

Quote: "it's better /kernel/drivers/net/wireless/realtek/rtl8821cu as "dest module location""

Reply: I'm all ears. Convince me why this is so.

**Comment by Digitalone1** [comment]
> Quote: "it's better /kernel/drivers/net/wireless/realtek/rtl8821cu as "dest module location""
> 
> Reply: I'm all ears. Convince me why this is so.

It's more specific. In example, inside `/usr/lib/modules/5.12.8-zen1-1-zen/kernel/drivers/net` on my system I have various folders related to brands: broadcom, intel, mediatek, realtk, etc.

I suppose this is intended to have a better organization of modules. I have another dkms module, the ethernet [r8168](https://aur.archlinux.org/packages/r8168-dkms/) and this is located in `/kernel/drivers/net/ethernet/realtek`. See [here](https://aur.archlinux.org/cgit/aur.git/tree/dkms.conf?h=r8168-dkms). 

It's not a big issue, but just for better organization of module location.

**Comment by Digitalone1** [comment]
Installed. It's working good.

I used the following custom `PKGBUILD` on Arch:

```
pkgname=rtl8821cu-dkms-git
_pkgbase=rtl8821cu
_pkgver="5.8.1.7"
pkgver=5.8.1.7.99.e54b8f4
pkgrel=1
pkgdesc="Realtek rtl8821cu chipset kernel driver"
arch=("x86_64")
url="https://github.com/morrownr/8821cu"
license=("GPL2")
depends=("dkms")
makedepends=("git" "bc")
conflicts=("${_pkgbase}" "${_pkgbase}-dkms-git")
provides=("${_pkgbase}")
source=(
    "git+https://github.com/morrownr/8821cu"
    "dkms.conf"
)
sha256sums=("SKIP" "SKIP")


pkgver() {
   cd ${srcdir}/8821cu
   printf "%s" ${_pkgver} "." "$(git rev-list --count HEAD)" "." "$(git rev-parse --short HEAD)"
}

package() {
    cd ${srcdir}/8821cu

    mkdir -p ${pkgdir}/usr/src/${_pkgbase}-${_pkgver}

    cp -rf * ${pkgdir}/usr/src/${_pkgbase}-${_pkgver}
    cp ${srcdir}/dkms.conf ${pkgdir}/usr/src/${_pkgbase}-${_pkgver}
}
```

with custom `dmks.conf`

```
PACKAGE_NAME="rtl8821cu"
PACKAGE_VERSION="5.8.1.7"
BUILT_MODULE_NAME[0]="8821cu"
DEST_MODULE_LOCATION[0]="/kernel/drivers/net/wireless/realtek/rtlwifi/rtl8821cu"
MAKE="'make' -j$(nproc) KVER=${kernelver} KSRC=/lib/modules/${kernelver}/build"
AUTOINSTALL="yes"
REMAKE_INITRD="no"
```

Will report any information in case of problems. No issues so far.

**Comment by morrownr** [comment]
Quote: "Edit: maybe you could only set /kernel/drivers/net/wireless/realtek"

Reply: I have made note of this and will consider it.

Glad the driver is working for you.

**Comment by Uefi1** [comment]
> Итак, нет никаких реальных различий по сравнению с [этим](https://aur.archlinux.org/packages/rtl8821cu-dkms-git/) ( [git](https://github.com/brektrou/rtl8821CU) )?
> 
> В любом случае, [пакет](https://aur.archlinux.org/packages/rtl8821cu-git/) для вашей версии на AUR не предназначен для установки dkms.
> 
> Myabe Я попробую установить его с помощью специального PKGBUILD, но сделаю небольшое предложение: лучше, `/kernel/drivers/net/wireless/realtek/rtl8821cu`если [здесь](https://github.com/brektrou/rtl8821CU/blob/master/dkms.conf) задано «расположение модуля dest» .

Currently the most stable and efficient driver on the 8811/8821cu from Morrownr  I have tested everything there is
